config.go

```diff
@@ -1917,6 +1925,52 @@ func parseRPCParams(cConfig *lncfg.Chain, nodeConfig interface{},
 	return nil
 }
 
+// extractEnvVariable extracts the value of an environment variable from a string.
+// It supports the following formats:
+// 1) $ENVIRONMENT_VARIABLE
+// 2) ${ENVIRONMENT_VARIABLE}
+// 3) ${ENVIRONMENT_VARIABLE:-DEFAULT_VALUE}
+//
+// Standard environment variable naming conventions:
+// - ENVIRONMENT_VARIABLE contains letters, digits, and underscores, and does not start with a digit.
+// - DEFAULT_VALUE follows the rule that it can contain any characters except whitespace.
+//
+// Parameters:
+// - value: The input string containing references to environment variables.
+//
+// Returns:
+//   - string: The value of the specified environment variable, or the default value
+//     if provided, or the original input string if no matching variable is found or set.
+func extractEnvVariable(value string) string {
+	re := regexp.MustCompile(`^(?:\${([a-zA-Z_][a-zA-Z0-9_]*)?(:-(\S+))?})|(?:\$([a-zA-Z_][a-zA-Z0-9_]*))$`)
+
+	matches := re.FindStringSubmatch(value)
+
+	if len(matches) > 0 {
```

Comment:
I'd combine these lines to make it such that the matches variable is only valid in the block scope.